docs/proposals/dynamic-openshift-settings-configmaps.md

```diff
@@ -0,0 +1,55 @@
+# Changing OpenShift settings dynamically through ConfigMaps
+
+## Problem
+
+Currently, there are settings contained withing the master-config.yaml file that
+are of value to an unpriviledged user. Specifically, the `autoProvisionEnabled`
+boolean under `jenkinsPipelineConfig` is only available to users with the
+ability to write to the master-config.
+
+## Design proposal:
+
+- In each project there can exist a ConfigMap annotated with
+  'build.openshift.io/configuration'
```

Comment:
Having the code handle having multiple configmaps at different places sounds a bit dangerous, I put a suggestion in for a utility function to consolidate multiple configmaps.